File: 1679-max-number-of-k-sum-pairs/solution.py
class Solution:
    # Solution 1
    def maxOperations(self, nums: List[int], k: int) -> int:
        a = {}
        operations = 0

        for _, i in enumerate(nums):
            e = k - i
            if e in a and a[e] > 0:
                a[e]-=1
                operations+=1
            else:
                if i not in a:
                    a[i] = 0
                a[i] += 1
        return operations
        

File: 1679-max-number-of-k-sum-pairs/solution.go
package main

func maxOperations(nums []int, k int) int {
	a := make(map[int]int)
	op := 0
	for _, i := range nums {
		e := k - i
		if a[e] > 0 {
			a[e]--
			op++
		} else {
			a[i]++
		}
	}
	return op
}
